rholang/examples/tut-prime.rho

```diff
@@ -0,0 +1,22 @@
+new loop, primeCheck in {
+  contract loop(@x) = {
+    match x {
+      [] => Nil
+      [head ...tail] => {
+        new ret in {
+          for (_ <- ret) {
+            loop!(tail)
+          } | primeCheck!(head, *ret)
+        }
+      }
+    }
+  } |
+  contract primeCheck(@x, ret) = {
+    match x {
+      Nil => @"stdoutAck"!("Nil", *ret)
+      ~{~Nil | ~Nil} => @"stdoutAck"!("Prime", *ret)
```

Comment:
Do you expect people to understand what `~{~Nil | ~Nil}` means? not a par of not `Nil`s ? How does it check that it's a prime?